dts/bindings/mtd/mxicy,mx25r-spi.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Properties for features only available on the Macronix Ultra Low Power
+  SPI NOR flash devices (MX25R*) using the Zephyr spi-nor flash driver.
+
+compatible: "mxicy,mx25r-spi"
```

Comment:
If we only rely on the `mxicy,high-perf-mode` property, we have no way to turn off high performance mode on those flash devices with a firmware update. Since then the only method of returning to ultra low power mode would be to cold boot the device. @mbolivar also wanted to keep the properties that aren't available for other devices out of the common binding.

Reply:
> `mxicy,mx25r-spi-nor` otherwise it looks like this is bus driver compatible.

I'm new to device tree bindings, but the docs said you can add the `on-bus` value to the binding name (unfortunately I'm not able to find back to that doc page now). So the `-spi` specifies the bus, not the device type. I think it's implicit that mx25r is a spi-nor device, since that device series is spi-nor, but they can also be used with qspi. The binding is only compatible with spi for now, but we can (in another PR) move the property out to a `mxicy,mx25r-common.yaml` and then include it from the `-spi.yaml` and `-qspi.yaml` if I've understood the bindings correctly.